apps/merkle_patricia_tree/lib/merkle_patricia_tree/caching_trie.ex

```diff
@@ -51,13 +72,36 @@ defmodule MerklePatriciaTree.CachingTrie do
   @impl true
   def remove_key(caching_trie, key) do
     key_nibbles = Helper.get_nibbles(key)
+    trie_change = {:remove, caching_trie.in_memory_trie.root_hash, key}
```

Comment:
I wonder if this could be extracted into a function we could chain on the end of `updated_caching_trie` i.e.

```ex
caching_trie
|> fetch_node()
|> Destroyer.remove_key(key_nibbles, caching_trie)
|> put_node(caching_trie)
|> into(caching_trie)
|> store()
|> add_to_trie_changes(caching_trie.in_memory_trie.root_hash, key)
```